torch/_export/passes/lift_constants_pass.py

```diff
@@ -197,20 +197,11 @@ def lift_constants_pass(
 
     first_user_input_loc, first_user_input = 0, next(iter(gm.graph.nodes))
     used_target_names = set()
-    for node in gm.graph.nodes:
-        if node.op == "placeholder":
-            if node.name in graph_signature.user_inputs:
-                first_user_input = node
-                break
-            used_target_names.add(inputs[first_user_input_loc].target)
-            first_user_input_loc += 1
-        # If we ever hit here, it means that
-        # there was no user input so the constants
-        # should be inserted right before the first
-        # non-placeholder node.
-        if node.op != "placeholder":
-            first_user_input = node
-            break
+
+    input_nodes = [node for node in gm.graph.nodes if node.op == "placeholder"]
+    assert len(input_nodes) == len(input_specs)
+    for node, input_spec in zip(input_nodes, input_specs):
+        used_target_names.add(input_spec.target)
```

Comment:
why do we need `input_nodes` in this for-loop here? Looks like we're just adding all input_spec's target